src/libsyntax/parse/parser.rs

```diff
@@ -5350,17 +5350,25 @@ impl<'a> Parser<'a> {
     // LT_BOUND = LIFETIME (e.g., `'a`)
     // TY_BOUND = TY_BOUND_NOPAREN | (TY_BOUND_NOPAREN)
     // TY_BOUND_NOPAREN = [?] [for<LT_PARAM_DEFS>] SIMPLE_PATH (e.g., `?for<'a: 'b> m::Trait<'a>`)
-    fn parse_generic_bounds_common(&mut self, allow_plus: bool) -> PResult<'a, GenericBounds> {
+    fn parse_generic_bounds_common(&mut self,
+                                   allow_plus: bool,
+                                   colon_span: Option<Span>) -> PResult<'a, GenericBounds> {
         let mut bounds = Vec::new();
+        let mut has_removed_colon = false;
+        let mut negative_bounds = Vec::new();
+        let mut last_plus_span = None;
+        let mut promoted_plus_span = None;
```

Comment:
```suggestion
        let mut new_colon_span = None;
```

This should read better